### ISSUE TITLE
concat_ws and list aggregate

### ISSUE THREAD

**Issue by shishkin** [duckdb/duckdb#11773]
### What happens?

It seems that `concat_ws` doesn't work when wrapping over a `list` aggregation function. Am I missing something?

### To Reproduce

Tried it in a jupyter notebook:

```sql
select concat_ws(', ', list(a order by a desc)) from (select * as a from unnest(['1','2','3']))
```

Produces `[3, 2, 1]` as a list.

The same in DuckDB CLI:

```
D select concat_ws(', ', list(x order by x desc)) from (select unnest([1,2,3]) as x);
┌────────────────────────────────────────────────────────────────────┐
│ concat_ws(', ', list_sort(list(x), 'DESCENDING', 'ORDER_DEFAULT')) │
│                              varchar                               │
├────────────────────────────────────────────────────────────────────┤
│ [3, 2, 1]                                                          │
└────────────────────────────────────────────────────────────────────┘
```

### OS:

MacOS aarch64

### DuckDB Version:

0.10.2

### DuckDB Client:

Python and CLI

### Full Name:

Sergey Shishkin

### Affiliation:

shishkin.org

### What is the latest build you tested with? If possible, we recommend testing with the latest nightly build.

I have tested with a stable release

### Did you include all relevant data sets for reproducing the issue?

Yes

### Did you include all code required to reproduce the issue?

- [X] Yes, I have

### Did you include all relevant configuration (e.g., CPU architecture, Python version, Linux distribution) to reproduce the issue?

- [X] Yes, I have

**Comment by soerenwolfers** [comment]
This is as expected. You're calling `concat_ws(x, y)` and are getting `y::VARCHAR`. You seem to think you're calling `concat_ws(x, y[0], ..., y[n])`

**Comment by soerenwolfers** [comment]
To get what you want, you could use `list_reduce(list(...), (x, y) -> concat_ws(', ', x, y))` although I feel like there might be a more elegant solution. As far as I'm aware there is no unpacking operator (like Python's star operator) though.

**Comment by cmdlineluser** [comment]
Perhaps [array_to_string()](https://duckdb.org/docs/sql/functions/nested#array_to_string)

**Comment by soerenwolfers** [comment]
@cmdlineluser  Ah, nice. TIL: 
- *All* aggregate functions can be used on lists via `list_aggregate(listValue, '<NAMEOFAGGREGATEFUNCTION>', <MOREARGSTOAGGREGATEFUNCTION>)`. For *most* aggregate functions there is a shorthand `list_<NAMEOFAGGREGATEFUNCTION>`. See https://duckdb.org/docs/sql/functions/nested#list-aggregates
- There is a regular aggregate function called `string_agg(..., sep)`.  See https://duckdb.org/docs/sql/aggregates.html. It's one of the aggregates that has a list shorthand: `list_string_agg(..., sep)` (like all aggregate functions it can also be called via `list_aggregate(..., 'string_agg', sep)` on lists)
- For some reason, `list_string_agg` is also available as `array_to_string`

**Comment by szarnyasg** [comment]
@shishkin one piece of info - in IPython, I get a varchar too:

```ipython
In [1]: import duckdb

In [2]: duckdb.sql("select concat_ws(', ', list(x order by x desc)) from (select unnest([1,2,3]) as x);").show()
┌────────────────────────────────────────────────────────────────────┐
│ concat_ws(', ', list_sort(list(x), 'DESCENDING', 'ORDER_DEFAULT')) │
│                              varchar                               │
├────────────────────────────────────────────────────────────────────┤
│ [3, 2, 1]                                                          │
└────────────────────────────────────────────────────────────────────┘
```

**Comment by shishkin** [comment]
Thanks all for prompt explanations and suggestions. I ended up using `string_agg`, which works.

I don't think I like to variadic `concat_ws` and would prefer `concat_ws(sep, list)` with SQL engine complaining if the second argument is not a list. Also automatic coercion to string is error-prone.

**Comment by soerenwolfers** [comment]
I agree with both, but both are vestiges of older SQL systems and duckdb makes an effort to be compatible with those.

**Comment by shishkin** [comment]
I see. Then it's working as intended I guess.
